src/libstore/filetransfer.cc

```diff
@@ -160,7 +160,7 @@ struct curlFileTransfer : public FileTransfer
                 result.bodySize += realSize;
 
                 if (!decompressionSink) {
-                    decompressionSink = makeDecompressionSink(encoding, finalSink);
+                    decompressionSink = makeDecompressionSink("", finalSink);
```

Comment:
A "none" decompression sink is superfluous and can be removed.

IIRC the reason we're doing our own decompression here is that curl didn't have brotli support or wasn't compiled with it. We may want to check in our configure script that curl has support for brotli and zstd.

Reply:
Why should we prefer brotli, an algo optimized for html to compress nars that's mostly binary files? And properly implemented servers should not use encodings not explicitly advertised by the client to send responses, lacking support for a compression algo or two is not that critical from my point of view.